receiver/httpcheckreceiver/scraper.go

```diff
@@ -23,50 +26,71 @@ var (
 )
 
 type httpcheckScraper struct {
-	client   *http.Client
+	clients  []*http.Client
 	cfg      *Config
 	settings component.TelemetrySettings
 	mb       *metadata.MetricsBuilder
 }
 
 // start starts the scraper by creating a new HTTP Client on the scraper
 func (h *httpcheckScraper) start(_ context.Context, host component.Host) (err error) {
-	h.client, err = h.cfg.ToClient(host, h.settings)
+	for _, target := range h.cfg.Targets {
+		client, clentErr := target.ToClient(host, h.settings)
+		if clentErr != nil {
+			err = multierr.Append(err, clentErr)
+		}
+		h.clients = append(h.clients, client)
+	}
 	return
 }
 
 // scrape connects to the endpoint and produces metrics based on the response
 func (h *httpcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
-	if h.client == nil {
+	if h.clients == nil || len(h.clients) == 0 {
 		return pmetric.NewMetrics(), errClientNotInit
 	}
 
-	now := pcommon.NewTimestampFromTime(time.Now())
-
-	req, err := http.NewRequestWithContext(ctx, h.cfg.Method, h.cfg.Endpoint, http.NoBody)
-	if err != nil {
-		return pmetric.Metrics{}, err
+	var wg sync.WaitGroup
+	wg.Add(len(h.clients))
+	var mux sync.Mutex
+
+	for idx, client := range h.clients {
+		go func(client *http.Client, idx int) {
```

Comment:
nit: i'd rename the vars in here to avoid the confusion with the vars outside the func